lib/cache/collections.go

```diff
@@ -45,9 +46,9 @@ type collection interface {
 	watchKind() types.WatchKind
 }
 
-// executor[R] is a specific way to run the collector operations that we need
-// for the genericCollector for a generic resource type R.
-type executor[R types.Resource] interface {
+// executor[R, RD] is a specific way to run the collector operations that we need
+// for the genericCollector for a generic resource type R and its reader type RD.
+type executor[R types.Resource, RD any] interface {
```

Comment:
It was called G for "getter" in the RFD, but after looking at the actual code I decided to rename it to RD for "reader" (R is taken for "resource") because methods called `executor.getter()` and `collector.getter()` seemed confusing whether they return the getter or they are the getter. I wanted to clarify that by adding the verb "get", but it would be weird to call them `getGetter()`, so I figured "reader" was a good synonym. Naming _is_ hard 🙂

Reply:
T for Type of the resource and R for reader?

Reply:
💡 I like that! Renamed in https://github.com/gravitational/teleport/pull/24425/commits/e6a4b4a2a34e67fde4cf0e24b80f533cd5814030